### PR TITLE
Reimplement system app menu using MenuLayer.

### PR DESCRIPTION
**Pull request by zbsz**
Added generic Menu component.
Changed system app to use it.

### CODE REVIEW

**Comment by jwise** [comment]
Man is this change glorious.

File: Apps/System/menu.c [hunk 1]
```diff
@@ -1,222 +1,139 @@
...
-MenuItem _main_menu[4];
-static char *_selected_menu_name;
-static uint8_t _app_count = 0;
-
-#define MENU_MAIN       0
-#define MENU_WATCH      1
-#define MENU_CONSOLE    2
+MenuItems* menu_items_create(uint16_t capacity)
+{
+    MenuItems *result = (MenuItems *) calloc(1, sizeof(MenuItems));
```
Comment (jwise) [inline Apps/System/menu.c]: We should mark this at some point to make sure we use the right allocator (I think there to be an app-heap allocator, right?) -- an 'XXX' would be okay for that, no need to change it right now.  (@ginge do we have that app-heap allocator for system apps yet?)

File: Apps/System/menu.h [hunk 2]
```diff
@@ -1,35 +1,65 @@
...
- * WITHOUT ANY WARRANTY; without even the implied warranty of 
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU 
- * Lesser General Public License for more details.
+ * Generic menu component.
  *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ * RebbleOS
+ *
+ * Author: Barry Carter <barry.carter@gmail.com>.
```
Comment (jwise) [inline Apps/System/menu.h]: you can add yourself here

File: Apps/System/menu.c (hunk 1 above)
Comment (zbsz) [inline Apps/System/menu.c]: I can use the correct one already if it's there, I've just seen this used in scroll_layer before, isn't calloc actually mapped somewhere?

File: Apps/System/menu.h (hunk 2 above)
Comment (zbsz) [inline Apps/System/menu.h]: this file was already here, so put original author
besides, I don't like leaving such traces personally (useless vanity ;))

**Comment by zbsz** [comment]
this solves: https://github.com/pebble-dev/RebbleOS/issues/10

File: Apps/System/menu.c (hunk 1 above)
Comment (jwise) [inline Apps/System/menu.c]: yeah I think this does work right now, but I think that eventually we're going to want to move this to the app heap; can you mark all the allocation sites here with 'XXX fix when we have an app heap'

**Comment by zbsz** [comment]
added comments for allocation

**Pull request merged**